source/wp-content/themes/wporg-parent-2021/functions.php

```diff
@@ -106,6 +107,114 @@ function prevent_arrow_emoji( $content ) {
 	return preg_replace( '/([←↑→↓↔↕↖↗↘↙])/u', '\1&#65038;', $content );
 }
 
+/**
+ * Merge the child theme's theme.json into the parent theme.json.
+ *
+ * Pull the parent theme's values for array settings out and merge them by slug
+ * with the values in the child theme. This prevents values in the child theme
+ * from blowing away the parent theme's settings.
+ *
+ * Additional settings are merged correctly, since they're objects (and merged
+ * by key).
+ *
+ * See https://github.com/WordPress/gutenberg/issues/40557.
+ *
+ * @param WP_Theme_JSON_Data $theme_json Parsed child theme.json.
+ *
+ * @return WP_Theme_JSON_Data The updated theme.json settings.
+ */
+function merge_parent_child_theme_json( $theme_json ) {
+	// Nothing to merge if this is not a child theme.
+	if ( get_template_directory() === get_stylesheet_directory() ) {
+		return $theme_json;
+	}
+
+	// Build a new theme.json object.
+	$new_data = array(
+		'version' => 2,
+	);
+
+	$child_theme = $theme_json->get_data();
+
+	if ( ! empty( $child_theme['settings'] ) ) {
+		$parent_theme_json_file = get_template_directory() . '/theme.json';
+		$parent_theme_json_data = wp_json_file_decode( $parent_theme_json_file, array( 'associative' => true ) );
+		$parent_theme           = new \WP_Theme_JSON_Gutenberg( $parent_theme_json_data );
+
+		// Get base theme.json settings.
+		$theme_settings = $parent_theme->get_settings();
+		$settings       = $child_theme['settings'];
```

Comment:
I think it would make it clearer further down if these were more specifically named
```suggestion
		$parent_settings = $parent_theme->get_settings();
		$child_settings  = $child_theme['settings'];
```

Reply:
Good catch — my first try at this used style variations, so it was theme settings vs a given variation, and I didn't change all the variables. Updated in https://github.com/WordPress/wporg-parent-2021/pull/73/commits/d607c11e976e7ac518cd5784d0c9e87025f8f1f4.